src/components/ReviewTransaction.tsx

```diff
@@ -73,6 +79,58 @@ export function ReviewSummaryItem(props: {
   )
 }
 
+export function ReviewSummaryItemContact({
+  testID,
+  header,
+  recipient,
+}: {
+  testID?: string
+  header: string
+  recipient: Recipient
+}) {
+  const contact = useMemo(() => {
+    const phone = recipient.displayNumber || recipient.e164PhoneNumber
+    if (recipient.name) {
+      return { title: recipient.name, subtitle: phone, icon: UserIcon, testID: 'Name' }
+    }
+
+    if (phone) {
+      return { title: phone, icon: PhoneIcon, testID: 'Phone' }
+    }
+
+    if (recipient.address) {
+      return { title: recipient.address, icon: WalletIcon, testID: 'Address' }
+    }
+  }, [recipient])
+
+  // Should never happen
+  if (!contact) {
+    Logger.error(
+      'ReviewSummaryItemContact',
+      `Transaction review could not render a contact item for recipient header: ${header}`
```

Comment:
if `header` is always "to", then this error probably doesn't need to reference the `header` at all :)